### PR TITLE
feat: make sources extensible

### PR DESCRIPTION
**Pull request by fbiville**
The spec does not predefine any source anymore.

Users must now provide their own source implementations.

As a consequence, the JSON schema around sources is now limited to name and type.

Existing source-related validators have been removed as they also need to be user-provided now.

### CODE REVIEW

**Comment by neo4j-connectors** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: Please include a description of your PR changes.</td>
    </tr>
  </tbody>
</table>


> Pull request should have a description of the underlying changes.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1c57b669ee19022e97a70851d910ca0b74444c04
</p>

File: src/test/java/org/neo4j/importer/v1/sources/TextSource.java [hunk 1]
```diff
@@ -19,35 +19,47 @@
...
 
-public abstract class TextSource extends Source {
+public abstract class TextSource implements Source {
 
-    private final List<String> header;
+    private String name;
+
+    private List<String> header;
+
+    TextSource() {}
```
Comment (ali-ince) [inline src/test/java/org/neo4j/importer/v1/sources/TextSource.java:28]: we no longer need this default constructor, right?

**Review by ali-ince — COMMENTED** [review]
It looks good to me 💯

File: src/test/java/org/neo4j/importer/v1/sources/TextSource.java (hunk 1 above)
Comment (fbiville) [reply src/test/java/org/neo4j/importer/v1/sources/TextSource.java]: fixed now

**Review by ali-ince — APPROVED** [review]
LGTM

**Pull request merged**